### PR TITLE
Consider CUDS upon keyword generation. Close #34.

### LINKED ISSUE #34: Generator for keywords need to read simphony_metadata.yml
Currently the 'keywords' command of the generator only takes the cuba.yml and not the simphony_metadata.yml, this leads to missing CUBA keywords as seen in https://github.com/simphony/simphony-common/pull/300/files

### PR DESCRIPTION
**Pull request by mehdisadeghi**
@kitchoi Please see my changes to the generator.

### CODE REVIEW

File: simphony_metadata/scripts/generate.py [hunk 1]
```diff
@@ -29,8 +27,8 @@
 # FIXME: These are excluded because they are not defined CUBA keys
 EXCLUDE_SUPPORTED_PARAMETERS = ('definition', 'models', 'variables', 'data',)
 
-# validation.py for validation codes
-VALIDATION_PY_PATH = os.path.splitext(validation.__file__)[0]+'.py'
+# validation.py for validation codes.
+VALIDATION_PY_PATH = 'simphony_metadata/scripts/validation.py'
```
Comment (kitchoi) [inline simphony_metadata/scripts/generate.py]: Would this prevent the click command from being invoked in other directories?  Alternatively you can use `inspect` to get the path to this generate.py file and then compose the path to the validation.py

File: simphony_metadata/scripts/tests/test_validation.py [hunk 2]
```diff
@@ -96,6 +97,9 @@ def test_error_validate_cuba_keyword(self):
             else:
                 msg = ('Error is not raised for {cuba}: {value}'
                        '{cuba} should be a {type} with shape {shape}')
+                key = cuba_name
+                if key.startswith('CUBA.') or key.startswith('czba.'):
```
Comment (kitchoi) [inline simphony_metadata/scripts/tests/test_validation.py]: typo: `czba.`

File: simphony_metadata/scripts/tests/test_validation.py [hunk 3]
```diff
@@ -96,6 +97,9 @@ def test_error_validate_cuba_keyword(self):
             else:
                 msg = ('Error is not raised for {cuba}: {value}'
                        '{cuba} should be a {type} with shape {shape}')
+                key = cuba_name
```
Comment (kitchoi) [inline simphony_metadata/scripts/tests/test_validation.py]: `key = cuba_name.upper()` here?

**Comment by kitchoi** [comment]
@mehdisadeghi Please update the command in .travis.yml too

**Comment by kitchoi** [comment]
```
+import validation
```

Relative imports are discouraged and they will not be supported in Python3.
Why do you want to get rid of `from . import validation`?

**Comment by mehdisadeghi** [comment]
> Relative imports are discouraged

Is this a typo? Relative imports are ~~encouraged~~ **discouraged**, ~~right?~~

> Why do you want to get rid of from . import validation?

I don't want to get rid of ~~relative~~ **absolute** imports, I was just reverting some changes and putting things together so I missed this import.

**Comment by kitchoi** [comment]
> Is this a typo? Relative imports are encouraged, right?

I meant _implicit_ relative imports, yes it was a typo.

**Comment by mehdisadeghi** [comment]
@kitchoi Oh, my bad! You are right :)

**Comment by mehdisadeghi** [comment]
@kitchoi is this OK to merge?

**Comment by kitchoi** [comment]
@mehdisadeghi The codecov complains do not seem warranted.  Yes please merge.  Thanks.

**Pull request merged**